### PR TITLE
feat: added filter with categories and rating

### PR DESCRIPTION
**Pull request by aryan-13**
Features Added: 
- Sort - High to low/ low to high
- Filter by Categories
- Filter by Rating
 
Preview: 
https://deploy-preview-5--lunar-store.netlify.app/products

### CODE REVIEW

File: src/Components/FilterBar.jsx [hunk 1]
```diff
@@ -76,21 +97,25 @@ function FilterBar() {
...
+
 					<input
 						type="range"
 						list="tickmarks"
-						min="10"
-						max="50"
-						step="10"
+						onChange={(e) =>
+							filterDispatch({ type: 'RATING', payload: e.target.value })
+						}
```
Comment (anshikaa12) [inline src/Components/FilterBar.jsx:113]: is min max property required here?

File: src/Context/filter-context.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+};
+const sortByPrice = (productList, sortBy) => {
+	if (sortBy === 'LOW_TO_HIGH')
+		return productList.sort((a, b) => a.discountPrice - b.discountPrice);
+	else if (sortBy === 'HIGH_TO_LOW')
+		return productList.sort((a, b) => b.discountPrice - a.discountPrice);
+	else return productList;
+};
+
+const filterByCategory = (productList, filterCategories) => {
```
Comment (anshikaa12) [inline src/Context/filter-context.jsx:20]: you can move these functions in a separate file like helpers

**Review by anshikaa12 — APPROVED** [review]
suggested some changes, rest looks good!

File: src/Components/FilterBar.jsx [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
-	// useEffect(() => {
-	// 	getCatog();
-	// }, []);
+	const { filter, filterDispatch } = useFilter();
+	const [categories, setCategories] = useState();
+	const getCategory = async () => {
+		const categories = await axios.get('/api/categories');
+
+		try {
+			console.log('catogs: ', categories.data.categories);
```
Comment (lannisterrr) [inline src/Components/FilterBar.jsx:11]: remove consoles and use a code formatter tool  like preetier.

**Review by lannisterrr — COMMENTED** [review]
Some suggesstions made.

File: src/Components/FilterBar.jsx (hunk 3 above)
Comment (aryan-13) [reply src/Components/FilterBar.jsx:11]: I'll remove all of them once I'm done with the basic functionality

File: src/Components/FilterBar.jsx (hunk 1 above)
Comment (aryan-13) [reply src/Components/FilterBar.jsx:113]: Yes

File: src/Context/filter-context.jsx (hunk 2 above)
Comment (aryan-13) [reply src/Context/filter-context.jsx:20]: I'll refactor the code in another PR

**Pull request merged**